mypy/semanal.py

```diff
@@ -3546,6 +3550,11 @@ def visit_file(self, file: MypyFile, fnam: str, mod_id: str, options: Options) -
                 else:
                     typ = UnionType([UnboundType('__builtins__.str'),
                                      UnboundType('__builtins__.unicode')])
+            elif name == '__path__':
+                # __path__ is only defined on __init__.py files
+                if not os.path.basename(fnam).lower() == '__init__.py':
+                    continue
+                typ = UnboundType(t)
```

Comment:
Trying `typ = self.sem.named_type('builtins.list', [self.sem.named_type('builtins.str')])`, the test run raises `AssertionError: Name 'builtins' is not defined`. Similar exception for `__builtins__`.

Then trying `typ = UnboundType('__builtins__.list', args=[UnboundType('__builtins__.str')])`, several tests in the unit tests output error like: `error: "list" is not subscriptable, use "typing.List" instead`. Is `mypy`'s source allowed to to assume the presence of `typing.List` or is there a better way?